idp-service/src/test/java/io/mosip/idp/services/AuthorizationServiceTest.java

```diff
@@ -409,4 +378,182 @@ public void getOauthDetails_withValidClaimsInDbAndValidClaimsInReqAndNoOPENIDSco
             Assert.assertTrue(ex.getErrorCode().equals(ErrorConstants.INVALID_SCOPE));
         }
     }
+
+    @Test
+    public void authenticate_withInvalidTransaction_thenFail() {
+        String transactionId = "test-transaction";
+        when(cacheUtilService.getPreAuthTransaction(transactionId)).thenReturn(null);
+
+        KycAuthRequest kycAuthRequest = new KycAuthRequest();
+        kycAuthRequest.setTransactionId(transactionId);
+        try {
+            authorizationServiceImpl.authenticateUser(kycAuthRequest);
+            Assert.fail();
+        } catch (IdPException ex) {
+            Assert.assertTrue(ex.getErrorCode().equals(ErrorConstants.INVALID_TRANSACTION));
+        }
+    }
+
+    @Test
+    public void authenticate_multipleRegisteredAcrsWithSingleFactor_thenPass() throws IdPException, KycAuthException {
+        String transactionId = "test-transaction";
+        when(cacheUtilService.getPreAuthTransaction(transactionId)).thenReturn(createIdpTransaction(
+                new String[]{"mosip:idp:acr:generated-code", "mosip:idp:acr:static-code"}));
+
+        List<List<AuthenticationFactor>> allAuthFactors=new ArrayList<>();
+        allAuthFactors.add(getAuthFactors("mosip:idp:acr:generated-code"));
+        allAuthFactors.add(getAuthFactors("mosip:idp:acr:static-code"));
+        when(authenticationContextClassRefUtil.getAuthFactors(new String[]{"mosip:idp:acr:generated-code",
```

Comment:
why are we mocking  authenticationContextClassRefUtil full, can we not mock only the get mapping json and give json, so that is also covered ?

Reply:
AuthorizationServiceTest is not an integration test class. AuthenticationContextClassRefUtil is unit-tested here AuthContextClassRefUtilTest.java

Reply:
I am little confused here, if AuthorizationAPIFlowTest is a integration test, why is AuthorizationServiceTest is alone unit test ?

Reply:
what is the approach we are following in IdP ?

Reply:
Controllers and repositories are integration test cases and others are unit tests. 
Complete Authorization code flow i have created as one integration test class.